src/hotspot/share/gc/parallel/psOldGen.cpp

```diff
@@ -179,17 +179,17 @@ void PSOldGen::object_iterate_block(ObjectClosure* cl, size_t block_index) {
   }
 }
 
-HeapWord* PSOldGen::expand_and_cas_allocate(size_t word_size) {
-  expand(word_size*HeapWordSize);
+bool PSOldGen::expand_for_allocate(size_t word_size) {
+  bool result = expand(word_size*HeapWordSize);
   if (GCExpandToAllocateDelayMillis > 0) {
     os::naked_sleep(GCExpandToAllocateDelayMillis);
   }
-  return cas_allocate_noexpand(word_size);
+  return result;
 }
 
-void PSOldGen::expand(size_t bytes) {
+bool PSOldGen::expand(size_t bytes) {
   if (bytes == 0) {
-    return;
+    return true;
```

Comment:
I'd prefer if the code would `guarantee` or at least `assert` that `bytes > 0` because returning `true` here seems scary wrt to the loop.

All code paths seem to cover this situation already, i.e. with `word_size == 0` this should not be called.

But if you think it's not a big issue,

Reply:
Good point.  I will make sure a 0 size never gets here and assert/guarantee, or otherwise figure out what to do.

Reply:
I've changed various quick returns on zero size to instead be asserts, since none of them should ever be called with a zero size.